src/main/java/com/philips/research/bombase/core/nuget/domain/NugetClient.java

```diff
@@ -0,0 +1,103 @@
+    private static final ObjectMapper XML_MAPPER = new XmlMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.NON_PRIVATE);
+    private final NugetAPI restJson;
+    private final NugetAPI restXml;
+    private final URI baseURI;
+
+    NugetClient() {
+        this(URI.create("https://api.nuget.org/v3/"));
+    }
+
+    NugetClient(URI uri) {
+        final var retrofitJson = new Retrofit.Builder()
+                .baseUrl(uri.toASCIIString())
+                .addConverterFactory(JacksonConverterFactory.create(MAPPER))
+                .build();
+        final var retrofitXML = new Retrofit.Builder()
+                .baseUrl(uri.toASCIIString())
+                .addConverterFactory(JacksonConverterFactory.create(XML_MAPPER))
+                .build();
+        restJson = retrofitJson.create(NugetAPI.class);
+        restXml = retrofitXML.create(NugetAPI.class);
+        baseURI = uri;
+    }
+
+    Optional<PackageMetadata> getPackageMetadata(PackageURL purl) {
+        // Some NuGet repositories requires a lower case package name.
+        final var lowerCasedPurlName = purl.getName().toLowerCase(Locale.ROOT);
+        final var catalogResponse =
+                query(restJson.getCatalogEntry(lowerCasedPurlName, purl.getVersion()));
+
+        final var catalogEntryPath = catalogResponse.map(cat -> cat.extractCatalogEntryPath(cat, baseURI))
+                .orElseGet(Optional::empty);
+
+        if (catalogEntryPath.isPresent()) {
+            final var nugetSpecResponse =
+                    query(restXml.getNugetSpec(
+                            String.format("flatcontainer/%s/%s/%s.nuspec",
+                                    lowerCasedPurlName,
+                                    purl.getVersion(),
+                                    lowerCasedPurlName)));
+            return query(restJson.getDefinition(catalogEntryPath.get()))
+                    .map(result -> {
+                        result.downloadLocation =
+                                catalogResponse.get().packageContent;
+                        result.sourceUrl = nugetSpecResponse.map(s -> s.metadata.repository.url).orElse(null);
+                        return result;
+                    });
+        }
+        return Optional.empty();
+    }
+
+    private <T> Optional<T> query(Call<? extends T> query) {
+        try {
+            final var response = query.execute();
+            if (response.code() == 404) {
+                return Optional.empty();
+            }
+            if (!response.isSuccessful()) {
+                throw new NugetException("Nuget server responded with status " +
+                        response.code());
+            }
+            return Optional.ofNullable(response.body());
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("JSON formatting error", e);
```

Comment:
After testing... strangely enough, it will throw a JSONParseException. 
![image](https://user-images.githubusercontent.com/15904543/126508676-95519b53-726f-4fc6-83fc-89c5bde87608.png)
I will modify the message to also mention XML.